### PR TITLE
fix-web-freeze-header-module-bug-in-group-view

### PR DESCRIPTION
**Pull request by jlzhou**

### CODE REVIEW

File: max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
-;(function ($, window, undefined){
+function freeze_list_view_header(){
```
Comment (lasley) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: This isn't v9 JS API & is outside the inheritance chain. Take a look at https://github.com/OCA/maintainer-tools/pull/208/files for upgrade example

Comment (jlzhou) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: It is difficult to use standard v9 API on this case. I have less idea to get the code working. Can you help?

Comment (jlzhou) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: It is hard to attach freeze table header code onto 'reload_content's 'done' process of list_view, already been stuck in this 2 days.

`            self.reload_content().done(function() {
                var one2many_length = $('.o_form_field.o_form_field_one2many').length;
                var many2many_length = $('.o_form_field.o_form_field_many2many').length;
                var scrollArea = $(".o_content")[0];

                if(scrollArea && one2many_length == 0 && many2many_length == 0) {
                    $('table.o_list_view').each(function () {
                        $(this).stickyTableHeaders({scrollableArea: scrollArea});
                    });
                }
            });`

Comment (lasley) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: I think you want `ListView.extend(` instead

Comment (jlzhou) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: Do you mean I should create a new widget extent from list view, and then add fixed header ability on it?

Comment (lasley) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: My bad, I read it wrong. You're doing it right with the `include`.

I think you should be overloading the `reload_content` method, appending the `.done` to the return of super, then returning the promise. You're working with a different promise by recalling the method in `do_show`

Comment (jlzhou) [inline max_web_freeze_list_view_header/static/src/js/max_web_freeze_list_view_header.js]: I've changed the code as below, but it doesn't work on the first time data loading of list view:

    ListView.include({
        reload_content: function () {
            var self = this;
            var xxx = $.Deferred();
            this._super.apply(this, arguments).done(function() {
                alert('freeze!');
                var one2many_length = $('.o_form_field.o_form_field_one2many').length;
                var many2many_length = $('.o_form_field.o_form_field_many2many').length;
                var scrollArea = $(".o_content")[0];

                if(scrollArea && one2many_length == 0 && many2many_length == 0) {
                    $('table.o_list_view').each(function () {
                        $(this).stickyTableHeaders({scrollableArea: scrollArea});
                    });
                }

                xxx.resolve();
            });
            return xxx.promise();
        },
    });

**Pull request closed without merging**